shared/modules/Numeric.ts

```diff
@@ -0,0 +1,621 @@
+    return new Numeric(
+      this.value.add(alignOperandDenominations(this, numeric).value),
+      this.base,
+      this.denomination,
+    );
+  }
+
+  /**
+   * TODO: make it possible to subtract ETH - GWEI value. So if you have
+   * Numeric 1 with denomination ETH and Numeric 2 with Denomination WEI,
+   * first convert Numeric 2 to ETH then subtract the amount from Numeric 1.
+   *
+   * @param value
+   * @param base
+   * @param denomination
+   */
+  minus(
+    value: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    const numeric = Numeric.from(value, base, denomination);
+
+    return new Numeric(
+      this.value.minus(alignOperandDenominations(this, numeric).value),
+      this.base,
+      this.denomination,
+    );
+  }
+
+  times(
+    multiplier: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    const multiplierNumeric = Numeric.from(multiplier, base, denomination);
+    return new Numeric(
+      this.value.times(
+        alignOperandDenominations(this, multiplierNumeric).value,
+      ),
+      this.base,
+      this.denomination,
+    );
+  }
+
+  /**
+   * Divides the Numeric by another supplied Numeric, carrying over the base
+   * and denomination from the current Numeric.
+   *
+   * @param divisor - The Numeric to divide this Numeric by
+   * @param base
+   * @param denomination
+   * @returns A new Numeric that contains the result of the division
+   */
+  divide(
+    divisor: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    return new Numeric(
+      this.value.div(
+        alignOperandDenominations(
+          this,
+          Numeric.from(divisor, base, denomination),
+        ).value,
+      ),
+      this.base,
+      this.denomination,
+    );
+  }
+
+  greaterThan(
+    comparator: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    return this.value.greaterThan(
+      Numeric.from(comparator, base, denomination).value,
+    );
+  }
+
+  greaterThanOrEqualTo(
+    comparator: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    return this.value.greaterThanOrEqualTo(
+      Numeric.from(comparator, base, denomination).value,
+    );
+  }
+
+  lessThan(
+    comparator: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    return this.value.lessThan(
+      Numeric.from(comparator, base, denomination).value,
+    );
+  }
+
+  lessThanOrEqualTo(
+    comparator: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    return this.value.lessThanOrEqualTo(
+      Numeric.from(comparator, base, denomination).value,
+    );
+  }
+
+  isNegative() {
+    return this.value.isNegative();
+  }
+
+  isPositive() {
+    return this.isNegative() === false;
+  }
```

Comment:
I was thinking, if `this.value` is 0, it would fall into `isNegative`, we should priority 0 in this situation~

Reply:
Why would 0 be negative?

Reply:
Sorry actually `this.value` would result in `isPositive` in this case.

Reply:
Which would be what we expect right?

Reply:
But 0 is not positive or negative, it should be 0 🧐

Reply:
Discussion moved to #17324 '